lib/Types.cpp

```diff
@@ -0,0 +1,123 @@
+﻿/*
+ * Copyright 2022-2024 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <limits>
+
+#include "streaming_protocol/Defines.h"
+#include "streaming_protocol/Types.h"
+
+namespace daq::streaming_protocol {
+
+Range::Range()
+{
+    clear();
+}
+
+bool Range::operator==(const Range &other) const
+{
+    return (
+            (std::abs(low - other.low) < epsilon) &&
+            (std::abs(high - other.high) < epsilon)
+           );
+}
+
+void Range::clear()
+{
+    low = -std::numeric_limits<double>::max();
```

Comment:
Are we sure that this works? Is the value "symmetric"?

Reply:
My idea is to have the possiblity to disable the limit. This would work with many types of numbers. Boolean and unsigned would need special handlings.

For boolean, there is no usefull limit at all. For unsigned types, the lower value for unlimited would be 0.

Reply:
Its just that for signed types the max and min are different. e.g. -128 - 127
the two values are not just the negative of the other, although -max would work. 
I was wondering if there would be a similar problem for a double

Reply:
Ah, of course it would work. low = -max and high = +max tell that one or both of limits are disabled. low  and high can have any value.  There should be only a validation on the server side that prohibits setting low >= high

Reply:
Right